src/applications/appeals/shared/validations/date.js

```diff
@@ -30,5 +34,59 @@ export const isInvalidDateString = (year, day, month, dateString) => {
   );
 };
 
-export const hasErrorParts = (errorParts, invalidDate) =>
-  errorParts.month || errorParts.day || errorParts.year || invalidDate;
+export const hasdatePartErrors = (datePartErrors, invalidDate) =>
+  datePartErrors.month ||
+  datePartErrors.day ||
+  datePartErrors.year ||
+  invalidDate;
+
+export const dateFunctions = rawString => {
+  const dateString = fixDateFormat(rawString);
+  const { day, month, year } = parseISODate(dateString);
+  const date = moment(rawString, FORMAT_YMD);
+  // get last day of the month (month is zero based, so we're +1 month, day 0);
+  // new Date() will recalculate and go back to last day of the previous month
+  const maxDays = year && month ? new Date(year, month, 0).getDate() : 31;
+  const invalidDate = dateString?.length < FORMAT_YMD.length || !date.isValid();
+  const datePartErrors = buildDatePartErrors(month, day, year, maxDays);
+  const todayOrFutureDate = date.isSameOrAfter(moment().startOf('day'));
+
+  return {
+    invalidDate,
+    datePartErrors,
+    isInvalidDateString: validateDateString(year, day, month, invalidDate),
```

Comment:
Please replace the `invalidDate` with `dateString`. `invalidateDate` is a boolean. This looks like it might be a bug in the original code.

Reply:
It might be a good idea to add `PropTypes` for all these functions.

Reply:
With it not being a component, we can't add PropTypes since these are functions.